File: BookApp/backend/src/main/java/com/bookapp/model/Comment.java
package com.bookapp.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "comments")
public class Comment {

    @Id
    private String id;

    private String userId;
    private String bookId;

    private String content;

    private LocalDateTime createdAt;
}

File: BookApp/app/src/main/java/com/bookapp/data/model/Comment.kt
package com.bookapp.data.model

data class Comment(
    val id: String? = null,
    val userId: String? = null,
    val username: String? = null,
    val fullName: String? = null,
    val avatar: String? = null,
    val bookId: String? = null,
    val content: String? = null,
    val createdAt: String? = null
)
